oif/productivity.py

```diff
@@ -0,0 +1,532 @@
+      'OHM_liquidity': next(filter(lambda tup: tup[0] == 'OHM', zip(symbols, balances)))[1]
+    } for id_, symbols, balances in zip(id_, symbols, balances)]
+
+  return pd.DataFrame(data)
+
+def token_liquidity(token_address: str, timestamp: Optional[int] = None) -> pd.DataFrame:
+  """ Returns a dataframe containing the following columns: Protocol, Address, Symbol, OHM_liquidity
+  and USD_liquidity. The OHM_liquidity is measured at timestamp :attr:`timestamp`.
+  """
+  token_price = get_price_token_address(token_address, timestamp)
+
+  if timestamp is not None:
+    block_number = block_of_timestamp(timestamp)
+    current_block = web3.eth.get_block_number()
+    
+    if block_number > current_block:
+      block_number = current_block - 5  # Subtract a couple blocks to account for graph-node delay
+  else:
+    block_number = None
+
+  full_df = pd.concat([
+    uniswap_v2_liquidity(token_address, block_number),
+    uniswap_v3_liquidity(token_address, block_number),
+    sushiswap_liquidity(token_address, block_number),
+    balancer_v2_liquidity(token_address, block_number),
+  ])
+
+  # Add USD value of OHM liquidity
+  full_df['USD_liquidity'] = full_df['OHM_liquidity'] * token_price
+
+  return full_df.set_index(['Protocol', 'Address', 'Symbol'])
+
+def token_liquidity_MoM(token_address: str, month: int, year: int):
+  (start, end) = get_month_start_end_timestamp(year, month)
+
+  # If current month, set end timestamp to None (to get current data)
+  if month == date.today().month and year == date.today().year:
+    end = None
+    
+  # Get months string labels
+  (previous_month_str, current_month_str) = (
+    month_name_map[12 if month == 1 else month-1],
+    month_name_map[month]
+  )
+
+  # Get data for previous month
+  df0 = token_liquidity(token_address, start).rename(columns={
+    'OHM_liquidity': 'OHM_liq_{}'.format(previous_month_str),
+    'USD_liquidity': 'USD_liq_{}'.format(previous_month_str),
+  })
+
+  # Get data for target month
+  df1 = token_liquidity(token_address, end).rename(columns={
+    'OHM_liquidity': 'OHM_liq_{}'.format(current_month_str),
+    'USD_liquidity': 'USD_liq_{}'.format(current_month_str),
+  })
+
+  # Calculate MoM change
+  df = pd.concat([df0, df1], axis=1)
+  df['OHM_liq_{}'.format(previous_month_str)].fillna(0.0, inplace=True)
+  df['USD_liq_{}'.format(previous_month_str)].fillna(0.0, inplace=True)
+  df['OHM_liq_{}'.format(current_month_str)].fillna(0.0, inplace=True)
+  df['USD_liq_{}'.format(current_month_str)].fillna(0.0, inplace=True)
+  df['OHM_liq_MoM_change'] = df['OHM_liq_{}'.format(current_month_str)] - df['OHM_liq_{}'.format(previous_month_str)]
+  df['USD_liq_MoM_change'] = df['USD_liq_{}'.format(current_month_str)] - df['USD_liq_{}'.format(previous_month_str)]
+  df['OHM_liq_MoM_change_percent'] = df['OHM_liq_MoM_change'] / df['OHM_liq_{}'.format(previous_month_str)]
+  df['USD_liq_MoM_change_percent'] = df['USD_liq_MoM_change'] / df['USD_liq_{}'.format(previous_month_str)]
```

Comment:
Was there a specific reason why you used `.format` here. I think it's nit picky for me to criticize but I saw you use `f` strings earlier so I was wondering if there was a specific reason or if it was just easier to duplicate lines here.